sos/plugins/fwupd.py

```diff
@@ -0,0 +1,51 @@
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+
+from sos.plugins import Plugin, RedHatPlugin, DebianPlugin, UbuntuPlugin
+from glob import glob
+
+
+class Fwupd(Plugin, RedHatPlugin, DebianPlugin, UbuntuPlugin):
+    """fwupd information
+    """
+
+    plugin_name = 'fwupd'
+    profiles = ('system', )
+
+    packages = ('fwupd',)
+
+    def setup(self):
+        self.add_cmd_output([
+            "fwupdmgr get-approved-firmware",
+            "fwupdmgr get-devices --no-unreported-check",
+            "fwupdmgr get-history",
+            "fwupdmgr get-remotes",
+        ])
+
+        self.add_copy_spec([
+            "/etc/fwupd",
+        ])
+
+        self.add_journal(units="fwupd")
+
+    def postproc(self):
+        for remotes_conf in glob("/etc/fwupd/remotes.d/*"):
+            self.do_file_sub(
```

Comment:
Isn't it easier to call:

    self.do_path_regex_sub("/etc/fwupd/remotes.d/*", r"Password=(.*)", r"Password=********")

? Not sure now if this will work for subdirectories as well, but I dont expect `/etc/fwupd/remotes.d/foo/bar.conf` configs to exist?

Reply:
> Not sure now if this will work for subdirectories as well

There isn't expected to be be subdirs in `remotes.d`